### ISSUE TITLE
Make `fink cleanup --all` clean harder (or add a `fink cleanup --everything` option?)

### ISSUE THREAD

**Issue by cooljeanius** [fink/fink#220]
I would like an option to make fink clean up even more things than `fink cleanup --all` currently does: specifically, unfinished builds, and all source tarballs. This would help save me a lot of disk space.

**Comment by dmacks** [comment]
See also https://sourceforge.net/p/fink/feature-requests/221/

**Comment by cooljeanius** [comment]
> I would like an option to make fink clean up even more things than `fink cleanup --all` currently does: specifically, unfinished builds, and all source tarballs. This would help save me a lot of disk space.

...to give some more specific numbers, the disk space that this would save for me on my current system is approximately 20.56GB.

**Comment by beren12** [comment]
I set my build dir to /tmp/ so it all vanishes on reboot :-) I also tend to keep my tarballs because there have been a few cases where the old ones vanish, but that's just me. My /opt/sw/src is 7.4gb right now, nothing for a 2tb nvme :-) (so glad there are adapters to let you use normal nvme with the older intel machines)

**Comment by dmacks** [comment]
The idea for the fink-implemented features was things that couldn't be done easily by hand, such as operating on the the internal dpkg database or using fink package data. If you want to just nuke /opt/sw/src *even in-use tarballs and abandoned builddirs*, that's just 'rm -rf'. I put my builddir in /opt/sw/build.build so I can control it separately from tarballs. But I can see a use-case for someone who doesn't know about all these fancy pathnames or cool things one can accomplish by changing them. For actions that could affect packages still "live" in the databases, probably best to use buildlock to avoid interfering with any packages that actually are in the process of being built.

**Comment by cooljeanius** [comment]
> The idea for the fink-implemented features was things that couldn't be done easily by hand, such as operating on the the internal dpkg database or using fink package data. If you want to just nuke /opt/sw/src _even in-use tarballs and abandoned builddirs_, that's just 'rm -rf'. 

Yeah I always feel a bit hesitant about `rm -rf`-ing anything; having it supported as a built-in `fink` command would help give me a bit more peace of mind that I'm actually allowed to do so.

**Comment by cooljeanius** [comment]
So yeah #258 looks like a good step towards addressing this
